flixel/tile/FlxBaseTilemap.hx

```diff
@@ -1296,4 +1371,28 @@ abstract FlxTilemapDiagonalPolicy(Int)
 	 * Diagonal movement costs one more than orthogonal movement
 	 */
 	var WIDE = 2;
-}
\ No newline at end of file
+}
+
+enum FlxTilemapPathingMethod
+{
+	/**
+	 * Basic method is calculating distances using an array covering the whole map (was original solution)
+	*/
+	BASIC;
+	/**
+	 * Optimization of basic method where distances are only calculated in a zone around start/end tiles
+	 * - scale > 1.0 determines how this rectangular zone is grown around the rectangle formed by start-end points
+	 * - minHorizonSize is used if any the scaled zone dimension is too small
+	*/
+	BASIC_SCALED_HORIZON(scale:Float, minHorizonSize:Int);
+}
+
+typedef FlxTilemapZone =
```

Comment:
Alternative solutions I can think of include (1) adding a member to each FlxBaseTilemap instance, (2) inline parameters, so instead of a zone object we have 6 parameters zoneX0, zoneY0, etc. (3) use a pooled object as you suggest or (4) we keep the current code using FlxTilemapZone.

A new dynamic object (on native target) will be created for each call of computePathDistance() and findPath()... If the impact of is an issue then I would suggest (2). What's your opinion on this?

Btw, you guys probably noticed but just to make sure: there is also an additional new in findPath() (var pathIndexes:Array = new Array();) I hope the runtime impact is acceptable.